### ISSUE TITLE
[enhancement]: Improvements to workflow list

### ISSUE THREAD

**Issue by skunkworxdark** [invoke-ai/InvokeAI#7255]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Contact Details

skunkworxdark

### What should this feature add?

Improvements to the recently added workflow list.  Originally reported in the Discord new-releases-discussion channel but I think they might have slipped under the radar.  https://discord.com/channels/1020123559063990373/1149506274971631688/1293478797953208382

1. You are not warned if you load a new workflow and you have unsaved changes to your current workflow. 
2. The default sort order of "Last opened" is not enabled when the UI is first loaded. 
3. When you load a workflow the sort order reverts to alphabetical (descending).
4. When you load a workflow, the Last opened order is stale and not updated by opening workflows until you refresh the UI.
5. There is no way of renaming a workflow unless, you load it, do a "save as" then delete the old one. 
6. There is no check/warning that you might be saving a workflow with the same name as an existing workflow.  It would be nice to warn you and allow you to overwrite the existing workflow rather than create a duplicate named workflow.

@maryhipp - acknowledge these in the discord so I guess is the best person to include here.

**Comment by maryhipp** [comment]
>There is no way of renaming a workflow unless, you load it, do a "save as" then delete the old one.
If you load a workflow, go to the Details tab and change the name, you should be able to just click the Save icon above that to save it in place. Let me know if that doesn't work for you.

![Screenshot 2024-11-04 at 2 24 20 PM](https://github.com/user-attachments/assets/4b899214-8879-4169-b74b-12b2d8900faa)

**Comment by skunkworxdark** [comment]
Yes, that works great for renaming.  I must have just missed this previously

**Comment by skunkworxdark** [comment]
A quick nudge on the sortation order of the workflows.  It is overall a fairly minor issue but if you use workflows much at all it quickly becomes annoying.
